src/app/shared/dimension.ts

```diff
@@ -63,6 +63,20 @@ export class Dimension {
                          this.time * other);
   }
 
+  log(other: number): Dimension {
```

Comment:
Please also add log and antilog variants to ScalarAndDimension. Note that a stub for `logEq` is already there and commented out.